cmd/conduit/cecdysis/decorators.go

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2025 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cecdysis
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/conduitio/conduit/cmd/conduit/api"
+	"github.com/conduitio/conduit/pkg/conduit"
+	"github.com/conduitio/ecdysis"
+	"github.com/spf13/cobra"
+)
+
+// ------------------- CommandWithClient
+
+// CommandWithExecuteWithClient can be implemented by a command that requires a client to interact
+// with the Conduit API during the execution.
+type CommandWithExecuteWithClient interface {
+	ecdysis.Command
+
+	// ExecuteWithClient is the actual work function. Most commands will implement this.
+	ExecuteWithClient(context.Context, *api.Client) error
+}
+
+// CommandWithExecuteWithClientDecorator is a decorator that adds a Conduit API client to the command execution.
+type CommandWithExecuteWithClientDecorator struct{}
+
+func (CommandWithExecuteWithClientDecorator) Decorate(_ *ecdysis.Ecdysis, cmd *cobra.Command, c ecdysis.Command) error {
+	v, ok := c.(CommandWithExecuteWithClient)
+	if !ok {
+		return nil
```

Comment:
just a question, when would this case happen? and why not return an error?

Reply:
@maha-hajja since this is a decorator, when not implemented by a command, we wouldn't want to run the code and it wouldn't imply an error. This is an expected behavior.

Reply:
These are the ecdysis decorators also for reference https://github.com/ConduitIO/ecdysis/blob/main/decorators.go